smart-contracts/scripts/lock/clone.js

```diff
@@ -0,0 +1,80 @@
+const { ethers } = require('hardhat')
+const createLock = require('../deployments/lock')
+const listManagers = require('./listManagers')
+
+async function main({
+  lockAddress,
+  unlockAddress,
+  unlockVersion,
+  serializerAddress,
+}) {
+  const PublicLock = await ethers.getContractFactory('PublicLock')
+  const lock = PublicLock.attach(lockAddress)
+
+  // eslint-disable-next-line no-console
+  console.log(`CLONE LOCK > serializing: ${await lock.name()}...`)
+
+  const Serializer = await ethers.getContractFactory('LockSerializer')
+  const serializer = Serializer.attach(serializerAddress)
+  const serializedLock = await serializer.serialize(lockAddress)
+
+  // create the new lock
+  const newLockAddress = await createLock({
+    serializedLock,
+    unlockAddress,
+    unlockVersion,
+    salt: web3.utils.randomHex(12),
+  })
+
+  // eslint-disable-next-line no-console
+  console.log('CLONE LOCK > add key owners...')
+  const newLock = PublicLock.attach(newLockAddress)
+  const { keyOwners, expirationTimestamps, keyManagers } = serializedLock
+  const keyTx = await newLock.grantKeys(
+    keyOwners,
+    expirationTimestamps,
+    keyManagers
+  )
+  const { events: keyEvents } = await keyTx.wait()
+  const transfers = keyEvents.filter(({ event }) => event === 'Transfer')
+  const keyManagersChanges = keyEvents.filter(
+    ({ event }) => event === 'KeyManagerChanged'
+  )
+  // eslint-disable-next-line no-console
+  console.log(
+    `CLONE LOCK > ${transfers.length} keys transferred, ${keyManagersChanges.length} key managers changed`
+  )
+
+  // eslint-disable-next-line no-console
+  console.log('CLONE LOCK > fetching managers...')
+
+  // fetch managers from graph
+  const managers = await listManagers({ lockAddress: newLockAddress })
+
+  // add back managers to the lock
+  if (managers.length) {
+    // eslint-disable-next-line no-restricted-syntax
+    const txs = Promise.all(
+      managers.map((manager) => newLock.addLockManager(manager))
+    )
+    const { events } = await txs.wait()
+    const added = events.map(({ event }) => event.name === 'LockManagerAdded')
+    added.forEach(({ args }) => {
+      // eslint-disable-next-line no-console
+      console.log(`LOCK CLONE > ${args[0]} added as lock manager.`)
+    })
+  }
```

Comment:
If more than 1 was added, let's also make sure we remove ourselves as lockManager.

Reply:
yep , this is more a generic task/script but this will be added in the locksmith route script